### PR TITLE
[TIMOB-25456] Android: Crash after calling JSON.stringify on Ti.UI.Window

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25456

**Description:**
Guard against accessing ActionBar reference before the inflating of the view.
Regression introduced in https://github.com/appcelerator/titanium_mobile/pull/9218

**Test case:**
app.js
```
var win = Ti.UI.createWindow();
Ti.API.error(JSON.stringify(win));
win.open();
```

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/proxy/ActionBarProxy.java [hunk 1]
```diff
@@ -55,7 +55,10 @@ public ActionBarProxy(AppCompatActivity activity)
 	{
 		super();
 		actionBar = activity.getSupportActionBar();
-		actionBar.setDisplayOptions(ActionBar.DISPLAY_USE_LOGO | ActionBar.DISPLAY_SHOW_HOME | ActionBar.DISPLAY_SHOW_TITLE);
+		// Guard against calls to ActionBar made before inflating the ActionBarView
+		if (actionBar != null) {
+			actionBar.setDisplayOptions(ActionBar.DISPLAY_USE_LOGO | ActionBar.DISPLAY_SHOW_HOME | ActionBar.DISPLAY_SHOW_TITLE);
+		}
```
Comment (garymathews) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/ActionBarProxy.java]: We should probably log a warning too if `actionBar` is null

**Review by garymathews — CHANGES_REQUESTED** [review]

**Comment by jquick-axway** [comment]
**Question:**
@ypbnv , will setting the action bar's title and logo still work after doing a `JSON.stringify()` on the window?
https://jira.appcelerator.org/browse/TIMOB-18872

**Comment by garymathews** [comment]
@jquick-axway it should do after `postlayout` ?

**Comment by ypbnv** [comment]
@garymathews Added a warning message, but I couldn't come up with anything better.

@jquick-axway According to the docs calls to ActionBar must be made after it's parent has dispatched an `open` event. 
_"In JavaScript, wait for the window or tab group's open event before accessing the action bar from the window or tab group's activity."_
From:
http://docs.appcelerator.com/platform/latest/#!/api/Titanium.Android.ActionBar

I tried the following code and it worked just fine:
```
var win = Ti.UI.createWindow();
Ti.API.error(JSON.stringify(win));
win.addEventListener('open', function() {
    win.activity.actionBar.setLogo('logo.png');
    win.activity.actionBar.setTitle('Title');
});
win.open();
```

File: android/titanium/src/java/org/appcelerator/titanium/proxy/ActionBarProxy.java [hunk 2]
```diff
@@ -55,7 +55,12 @@ public ActionBarProxy(AppCompatActivity activity)
 	{
 		super();
 		actionBar = activity.getSupportActionBar();
-		actionBar.setDisplayOptions(ActionBar.DISPLAY_USE_LOGO | ActionBar.DISPLAY_SHOW_HOME | ActionBar.DISPLAY_SHOW_TITLE);
+		// Guard against calls to ActionBar made before inflating the ActionBarView
+		if (actionBar != null) {
+			actionBar.setDisplayOptions(ActionBar.DISPLAY_USE_LOGO | ActionBar.DISPLAY_SHOW_HOME | ActionBar.DISPLAY_SHOW_TITLE);
+		} else {
+			Log.w(TAG, "Trying to get a reference to ActionBar before it's container was inflated.");
```
Comment (jquick-axway) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/ActionBarProxy.java]: Minor Issue:
The log message's word "it's" should be "its" without an apostrophe.

Comment (ypbnv) [reply android/titanium/src/java/org/appcelerator/titanium/proxy/ActionBarProxy.java]: Updated.

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Review by mukherjee2 — APPROVED** [review]
Passed FR with this environment:
Node Version: 6.11.5
NPM Version: 3.10.10
Mac OS: 10.13
Appc CLI: 7.0.0-master.8
Appc CLI NPM: 4.2.11-2
Titanium SDK version: 7.0.0 locally built
Appcelerator Studio vers 4.10.0
Android 7.1.2

**Pull request merged**